traffic_ops/testing/compare/compare.go

```diff
@@ -157,74 +159,301 @@ func testRoute(tos []*Connect, route string) {
+			} else if line[:len(SHORT_HEADER)] != SHORT_HEADER {
+				if len(line) >= len(LONG_HEADER) {
+					if line[:len(LONG_HEADER)] != LONG_HEADER {
+						scrubbedResult0 += line
 					}
 				} else {
-					refResult += refLine
+					scrubbedResult0 += line
 				}
 			}
 		}
 
-		for _, testLine := range testLines {
-			if len(testLine) < len(SHORT_HEADER) {
-				testResult += testLine
-			} else if testLine[:len(SHORT_HEADER)] != SHORT_HEADER {
-				if len(testLine) >= len(LONG_HEADER) {
-					if testLine[:len(LONG_HEADER)] != LONG_HEADER {
-						testResult += testLine
+		for _, line := range lines1 {
+			if len(line) < len(SHORT_HEADER) {
+				scrubbedResult1 += line
+			} else if line[:len(SHORT_HEADER)] != SHORT_HEADER {
+				if len(line) >= len(LONG_HEADER) {
+					if line[:len(LONG_HEADER)] != LONG_HEADER {
+						scrubbedResult1 += line
 					}
 				} else {
-					testResult += testLine
+					scrubbedResult1 += line
 				}
 			}
 		}
+	} else {
+		scrubbedResult0 = result0Str
+		scrubbedResult1 = result1Str
 	}
 
-	if refResult == testResult {
-		log.Printf("Identical results (%d bytes) from %s", len(res[0].Res), route)
+	if scrubbedResult0 == scrubbedResult1 {
+		log.Printf("Identical results (%d bytes) from %s\n", len(result0), route)
 	} else {
-		log.Print("Diffs from ", route, " written to")
-		for _, r := range res {
-			p, err := r.TO.writeResults(route, r.Res)
-			if err != nil {
-				log.Fatalf("Error writing results for %s: %s", route, err.Error())
+		writeAllResults(route, result0Str, (*responses)[0].TO, result1Str, (*responses)[1].TO)
+	}
+}
+
+// Given a slice of (exactly two) result objects, compares the JSON content of their responses
+// and write them to files if they differ. Ignores timestamps and Traffic Ops hostnames (to the
+// degree possible)
+func handleJSONResponse(responses *[]result, route string) {
+
+	// I avoid using `defer` to close the bodies because I want to do it as quickly as possible
+
+	result0Orig, err := ioutil.ReadAll((*responses)[0].Res.Body)
+	(*responses)[0].Res.Body.Close()
+	if err != nil {
+		(*responses)[1].Res.Body.Close()
+		log.Fatalf("Failed to read response body from %s/%s: %s\n", (*responses)[0].TO.URL, route, err.Error())
+	}
+	var result0 interface{}
+	if err = json.Unmarshal([]byte(result0Orig), &result0); err != nil {
+		log.Fatalf("Failed to parse response body from %s/%s as JSON: %s\n", (*responses)[0].TO.URL, route, err.Error())
+	}
+
+	result1Orig, err := ioutil.ReadAll((*responses)[1].Res.Body)
+	(*responses)[1].Res.Body.Close()
+	if err != nil {
+		log.Fatalf("Failed to read response body from %s/%s: %s\n", (*responses)[1].TO.URL, route, err.Error())
+	}
+	var result1 interface{}
+	if err = json.Unmarshal(result1Orig, &result1); err != nil {
+		log.Fatalf("Failed to parse response body from %s/%s as JSON: %s\n", (*responses)[1].TO.URL, route, err.Error())
+	}
+
+	// type-safety cast to maps
+	m0, m1 := result0.(map[string]interface{}), result1.(map[string]interface{})
```

Comment:
what happens if the cast fails?   wouldn't it be better to unmarshal directly to the map?

Reply:
unmarshaled JSON objects can always be cast to `map[string]interface{}` from what I understand. So yeah, I should be able to just unmarshal directly into the map.